### PR TITLE
Move CircleCI extensions to Marketplace - Part 2

### PR DESCRIPTION
**Pull request by gykuma**
Move CircleCI extensions (artifact, endpoint and task) to Marketplace

### CODE REVIEW

File: Extensions/CircleCI/Src/readme.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+This extension is an integration point for CircleCI&trade; with Release management in VS Team Services. With this extension, you can deploy artifacts from CircleCI&trade; builds using Release management. 
+
+**Note:** This extension work only with VS Team Services and TFS "15" RC onwards.
+
+## Usage
+This extension provides a service endpoint to connect to CircleCI&trade; account. Once connected, you can link a build artifact from the CircleCI&trade; project and deploy the same using Release management orchestration service.
+
+### Connecting to a CircleCI&trade; project
+Go to project settings -> Services tab and create a New Service Endpoint of type **CircleCI&trade":
```
Comment (omeshp) [inline Extensions/CircleCI/Src/readme.md]: fix

File: Extensions/CircleCI/Src/vss-extension.json [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+      "type": "ms.vss-distributed-task.task",
+      "targets": [
+        "ms.vss-distributed-task.tasks"
+      ],
+      "properties": {
+        "name": "Tasks/DownloadCircleCIArtifacts"
+        }
+    },
+    {
+      "id": "circleci-release-artifact-type",
```
Comment (omeshp) [inline Extensions/CircleCI/Src/vss-extension.json]: remove release

File: Extensions/CircleCI/Src/vss-extension.json [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+          {
+            "target": "artifacts",
+            "dataSourceName": "Artifacts",
+            "resultTemplate": "{ Value : \"{{{path}}}\", DisplayValue : \"{{path}}\", data: { itemType: \"file\" } }"
+          }
+        ],
+        "browsableArtifactTypeMapping": {
+          "folder": "artifactItems"
+        },
+        "yamlInputMapping": {
```
Comment (omeshp) [inline Extensions/CircleCI/Src/vss-extension.json]: remove for now.

File: Extensions/CircleCI/Src/vss-extension.json [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+        }
+    },
+    {
+      "id": "circleci-release-artifact-type",
+      "description": "i18n:CircleCI Artifact",
+      "_description.comment": "Don't Localize the word 'CircleCI'",
+      "type": "ms.vss-releaseartifact.release-artifact-type",
+      "targets": [
+        "ms.vss-releaseartifact.artifact-types",
+        "ms.vss-distributed-task.pipeline-artifact"
```
Comment (omeshp) [inline Extensions/CircleCI/Src/vss-extension.json]: remove for now.

**Review by omeshp — APPROVED** [review]

File: Extensions/CircleCI/Src/vss-extension.json [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+            "type": "ms.vss-endpoint.endpoint-auth-scheme-basic",
+            "inputDescriptors": [
+              {
+                "id": "username",
+                "name": "PAT",
+                "description": "Personal API Token",
+                "inputMode": "passwordbox",
+                "isConfidential": true,
+                "validation": {
+                  "isRequired": false,
```
Comment (PranaySPatil) [inline Extensions/CircleCI/Src/vss-extension.json]: isRequired should be true

File: Extensions/CircleCI/Src/vss-extension.json [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+                "name": "Password",
+                "inputMode": "none",
+                "isConfidential": true,
+                "validation": {
+                  "isRequired": false,
+                  "dataType": "string"
+                },
+                "values": {
+                  "inputId": "passwordInput",
+                  "defaultValue": ""
```
Comment (PranaySPatil) [inline Extensions/CircleCI/Src/vss-extension.json]: add isDisabled true

File: Extensions/CircleCI/Src/Tasks/DownloadCircleCIArtifacts/task.json [hunk 7]
```diff
@@ -0,0 +1,109 @@
+{
+    "id": "83751EE5-0AEF-4EC3-915F-DFB71B7157E7",
```
Comment (Lovakumar) [inline Extensions/CircleCI/Src/Tasks/DownloadCircleCIArtifacts/task.json]: preview = true?

File: Extensions/CircleCI/Src/readme.md [hunk 8]
```diff
@@ -0,0 +1,22 @@
+# CircleCI&trade; artifacts for Release management
+
+This extension is an integration point for CircleCI&trade; with Release management in VS Team Services. With this extension, you can deploy artifacts from CircleCI&trade; builds using Release management.
```
Comment (Lovakumar) [inline Extensions/CircleCI/Src/readme.md]: Change "VS Team Services" to Azure DevOps Services.

**Review by Lovakumar — APPROVED** [review]

File: Extensions/CircleCI/Src/readme.md (hunk 1 above)
Comment (gykuma) [reply Extensions/CircleCI/Src/readme.md]: fixed

File: Extensions/CircleCI/Src/Tasks/DownloadCircleCIArtifacts/task.json (hunk 7 above)
Comment (gykuma) [reply Extensions/CircleCI/Src/Tasks/DownloadCircleCIArtifacts/task.json]: made it preview

**Pull request merged**